### DISCUSSION TITLE
Discussion: Create zh-cn.cfg

### DISCUSSION THREAD

**Discussion by zer0Kerbal** [zer0Kerbal/KeridianDynamics#36] [General]
[@ShadowSTARS616](https://github.com/ShadowSTARS616) said:

> Great mod! In fact, this is my first attempt at localization, so there may be some shortcomings. In some places, such as part description, I added a little text to avoid new players' misunderstanding of the function of the part. If you think the text is inappropriate, please let me know and let us revise it together.

>
> I have tested the localization text in the game, and the translation of most parts works as expected. However, I found some problems, such as KD-FAVA, which I didn't found the corresponding localization identifier in en-en.cfg so that I couldn't add a translation to it. There's another sentence behind description of KD-ChemicalReactor, which localized identifier cannot be found.
Some combination words, like "keridian" and "keronica", I didn't quite understand their meaning, so I left them there as they were. Should I translate according to pronunciation?

**Comment by zer0Kerbal** [comment]
@ShadowSTARS616 
> Create zh-cn.cfg
> Great mod!
thank you - all the credit goes to the original author @EleusisLaArwall.

> In fact, this is my first attempt at localization, so there may be some shortcomings.
thank you @ShadowSTARS616 👍 very much appreciated.
_I didn't know you were creating a translation until very recently._

Kindly remember to create a PR (Push Request) into zer0Kerbal/KeridianDynamics so you get credit. Kindly push into the 0.8.99.0-prerelease branch. If you don't know how, not a big deal as I can edit your PR to some extent.

And shortcomings can be addressed, the lack of a translation is a bigger issue. :)

> In some places, such as part description, I added a little text to avoid new players' misunderstanding of the function of the part. If you think the text is inappropriate, please let me know and let us revise it together.
I trust your judgement. I can always update to address shortcomings or translations adjustments. ⌨️ 

> I have tested the localization text in the game, and the translation of most parts works as expected. However, I found some problems, such as KD-FAVA, which I didn't found the corresponding localization identifier in en-en.cfg so that I couldn't add a translation to it. There's another sentence behind description of KD-ChemicalReactor, which localized identifier cannot be found. 

Thank you, seems KD-FAVA was missed in the original localization pass. Now corrected.
here are the missing tags:
```
#KDVA-FAVA-titl = KD-FAVA
#KDVA-FAVA-desc = Provides the ability to assemble vessels anywhere without Kerbals. A LaunchSite is required in order to launch assembled vessels. The very high productivity factor can be used to improve production time on Launchpads.
#KDVA-FAVA-tags = base station assembl constru work produc build mobile (vab vehicle rocketparts
```
KS-ChemicalReactor - Please explain more.
`This high-end chemical reactor extracts almost all compounds from Ore. To achieve high yields the reactor needs to work slowly. The chemical extraction is the slow and efficient way to refine Ore.`

> Some combination words, like "keridian" and "keronica", I didn't quite understand their meaning, so I left them there as they were. Should I translate according to pronunciation?

I believe both of those are proper nouns, so translation might not be necessary. The origin of Keridian Dynamics, well either I don't know or never knew. Maybe @EleusisLaArwall will pop in and say. 📦 

KD-Keronica: I believe it is a remote construction probe; which allows remote crewless construction. I will look into it and adjust the description to be more specific.  Might be an allusion to [Majin Robo Veronica](https://powerrangers.fandom.com/wiki/Majin_Robo_Veronica)? Maybe @EleusisLaArwall will pop in and say. 📦 

also - can discuss here: https://github.com/zer0Kerbal/KeridianDynamics/discussions/36

**Reply by ShadowSTARS616** [reply]
> Kindly remember to create a PR (Push Request) into zer0Kerbal/KeridianDynamics so you get credit. Kindly push into the 0.8.99.0-prerelease branch. If you don't know how, not a big deal as I can edit your PR to some extent.

Actually I don't know how to do. Should I create a pull request at 0.8.99.0-prerelease branch page? Or just update it by Commit changes at the page of the file is acceptable? 
_Sorry, this is the first time I use Github to submit something instead of downloading. I have read the document, but I still don't understand how PR works. I searched push request but only found pages of pull request. It seems that push need a Github Desktop or used on command line?_

> KS-ChemicalReactor - Please explain more.
When I applied the localization, the description looks like this in the game: 
`这种高端化学反应器能从矿石中提取几乎所有的化合物。为了获得高产量，反应器需要缓慢工作。化学萃取是一种缓慢而有效的精炼方法。Can also extract RareMetals and ExoticMinerals from Dirt. Can also extract RareMetals and ExoticMinerals from Dirt.`

I used to think those extra texts were added by the mod PathManager, but after I uninstall it, these texts still exists. 

_I installed quite a lot of mod during the test, because I just thought that I could contribute translation for this mod when I was playing career mode. I have an exam tomorrow. After I finish the exam, I'll try to install only KeridianDynamics to see if this situation will happen again._

I will submit Taiwanese (zh-tw.cfg) translation at that time. 
In most cases, they are almost the same - the difference in the character is just like the difference between fonts, and the difference in use is like British English and American English. 

> Thank you, seems KD-FAVA was missed in the original localization pass. Now corrected.

KD-FAVA works correctly now. 👍

The description of KD-FabricationContainer doesn't work properly. I added  translated text to the file, but it doesn't work. 
_I forgot to said it. This problem existed when I first wrote localization files._

3D-printer O.G. is also not localized. Is this a part about to be discarded? 

> I believe both of those are proper nouns, so translation might not be necessary. The origin of Keridian Dynamics, well either I don't know or never knew. Maybe @EleusisLaArwall will pop in and say. 📦
KD-Keronica: I believe it is a remote construction probe; which allows remote crewless construction. I will look into it and adjust the description to be more specific. Might be an allusion to [Majin Robo Veronica](https://powerrangers.fandom.com/wiki/Majin_Robo_Veronica)? Maybe @EleusisLaArwall will pop in and say. 📦

Thank you for telling me what you know. They will be updated later when we have more information.

**Reply by zer0Kerbal** [reply]
All good. It wasn't long ago when I was asking those same questions. (Kaizen 改善) 😀 You are asking the right questions.

* try these - remember you have already 'forked' the original and 'committed' your changes to your own branch (`patch-1`)
  * https://medium.com/server-side-swift-and-more/a-simple-guide-to-creating-a-github-pull-request-48fee0afd015
  * https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request
  * https://learntocodetogether.com/create-your-first-pull-request/

from my Pull Request 
![image](https://user-images.githubusercontent.com/39887717/177399689-d3a74b36-7f57-41d9-8765-26c25e39da24.png)

there is (that I know of) no wrong way - and creating a PR does not harm anything. Plus Once you create it, I can modify most of it.

so start here:
![image](https://user-images.githubusercontent.com/39887717/177400083-9f64b673-36cc-44c4-b7fe-1bd265d52d5b.png)
then
![image](https://user-images.githubusercontent.com/39887717/177400302-49941d80-0be5-4213-9f0a-5393f9c3ee75.png)

then it is all about making it pull from your 'patch-1' into my '0.8.99.0-prerelease' branch

**Reply by zer0Kerbal** [reply]
> > KS-ChemicalReactor - Please explain more.
> > When I applied the localization, the description looks like this in the game:
> > `这种高端化学反应器能从矿石中提取几乎所有的化合物。为了获得高产量，反应器需要缓慢工作。化学萃取是一种缓慢而有效的精炼方法。Can also extract RareMetals and ExoticMinerals from Dirt. Can also extract RareMetals and ExoticMinerals from Dirt.`
> 
> I used to think those extra texts were added by the mod PathManager, but after I uninstall it, these texts still exists.
> 
> _I installed quite a lot of mod during the test, because I just thought that I could contribute translation for this mod when I was playing career mode. I have an exam tomorrow. After I finish the exam, I'll try to install only KeridianDynamics to see if this situation will happen again._

I understand now. There is a patch adding text to the part's description field. Something I will have to first find then address. Probably in the OSE Workshop patches. Good guess though concerning PatchManager.

Recently (as in last week) Module Manager was updated to allow for patching of localization strings "#". This will help. Have to think about how to best implement it. We can 'circle round back to it' soon.

---

> I will submit Taiwanese (zh-tw.cfg) translation at that time.
> In most cases, they are almost the same - the difference in the character is just like the difference between fonts, and the difference in use is like British English and American English.

✨ ! would be most appreciated.

Didn't realize that but makes sense. Most derived languages like Mexican Spanish or Brazilian Portuguese are that way. Didn't realize, but should have, that Taiwanese is a derived language.

---

> The description of KD-FabricationContainer doesn't work properly. I added translated text to the file, but it doesn't work.
> _I forgot to said it. This problem existed when I first wrote localization files._

thank you for the report - will look into it today. Which reminds me I need to create a bug report for the KD-FAVA which just fixed as well. Documentation, helpful and painful at the same time.

---

> 3D-printer O.G. is also not localized. Is this a part about to be discarded?

I have the unreleased '_experimental_' 0.9.0.0 release from @EleusisLaArwall, and have been working through implementing it.

The plan is to probably pull the NEW version until it can used as a new part; a larger (or smaller) sized version with matching stats and a different position on the tech tree.

There are other parts like this which I will be going through.

---

> Thank you for telling me what you know. They will be updated later when we have more information.

My pleasure!

Thank you for taking the time in the first place, along with your continued efforts.

zed'K

**Comment by zer0Kerbal** [comment]
@ShadowSTARS616 also you don't have to translate the following:
```
#KDVA-KDMobileVAB_ModuleColorChanger_toggleName = Toggle Lights
#KDVA-KDMobileVAB_ModuleColorChanger_eventOnName = Lights On
#KDVA-KDMobileVAB_ModuleColorChanger_eventOffName = Lights Off
```
or anything that has Toggle Lights/Lights On/Lights off. I am replacing those with localization strings scraped from KSP's internal dictionary, so it will be automatically translated into all languages which KSP speaks.

I also found that the `KD-FabricationContainer` has been localized.  However, I will keep my eyes open as I do a pass through all the parts. 

zed'K

**Comment by zer0Kerbal** [comment]
I am in the process of pushing out a new prerelease (0.8.99.0) and will have a new branch 0.8.99.1 ready after the release. @ShadowSTARS616

**Comment by zer0Kerbal** [comment]
@ShadowSTARS616 

Welcome back!

How is your Mother? Yourself?

zed'K

**Reply by ShadowSTARS616** [reply]
Thank you for your concern! 
Her condition has improved a lot since she was treated in the hospital. I need to accompany her back to the hospital for the next stage of treatment these days, and I will return to the college to continue my study at the end of August. 
Don't worry, we are all fine. 

I will continue to follow up the localization work, check whether the text needs to be updated, and modify those inappropriate places (if any).
